### ISSUE TITLE
Slim symbol usage in test output improved

### ISSUE THREAD

**Issue by fhoeben** [unclebob/fitnesse#984]
Currently when one uses a Slim symbol (which was previously assigned a value) in the test output one sees the symbol an arrow and the actual value. I would like this to be styled in such a way so that we don't always see both. (Please note that I'm not talking about the wiki page before the test is run, only about the output page, so the HTML can be changes without impacting the values sent to the test system.)

I would like to be able to get only the symbol OR the value (maybe with the other as `title` so it is shown as tooltip on hover?), so that the report becomes more concise and easier to read. (For symbol assignment a similar display approach would be great, but less important to me.) Ideally one could switch between the different representations (only value, only symbol, current) by clicking on the usage. 

Which representation should be shown initially is not immediately clear to me. For me, showing the current version for an assignment and only the symbol for usage would be best (as I have many usages of symbols having a large value which make the report unnecessarily big). This might be something to be controlled using a variable, or a setting in `plugins.properties`...

Any thoughts?

**Comment by fhoeben** [comment]
I prototyped a 'less advanced' approach in https://github.com/fhoeben/fitnesse/commit/3133931e0485f6c09e744bdd774d5901c1fb8d73

This uses the HTML tags details and summary to get a collapsable element (in modern browsers, older should just get all) and a title for the elements. I only show what was in the page by default, actual value is displayed/expanded on click (or shown as tooltip). I also added some classes so the display of the value/name may be improved using CSS.

**Comment by six42** [comment]
Hi @fhoeben ,

I like this a lot. 
I think your approach with using different HTML classes is the best approach as it gives flexibility to steer the appearance wit CSS on a global scale or collapse and expand on an individual level.
The question about the default is tricky and everybody might have a different opinion.
I think what you proposed as default is reasonable for most scenarios.

Maybe it is also an option to allow the test case writer to define the appearance of each variable with a "format" tag added to the symbol name.

Example:
$x      : current behaviour, display as "$x->lower"
$x#< :  collapsed, display as "$x"
$x#_  :  hide variable name show only the value, display as "lower"
$x#3 :  truncate display only the first three char, allow to expand to see the full text, display as $x->low...  
$x#100 :  truncate display only the first 100 char, display as $x->lower  
$x#/o.e/ :  regex display only what is matched by the regex /o.e/, allow to expand to see the full text, display as $x->...owe...

**Comment by fhoeben** [comment]
Allowing the format to be specified sounds line a desirable feature in itself. BUT that affects how existing wiki pages are evaluated, maybe someone has a page where a value of $x should be followed  by a literal '#<', so this would affect backwards compatibility.

If we are tackling the formatting, and therefore wiki representation of slim symbols, I would like to start with ensuring the syntax supports unification of the symbol's name begin and end (like in a scenario table where you can do `@var` but also `@{var}` so there is no ambiguity whether you meant parameter `va` followed by a literal `r`, or the parameter `var`). The fact that no such mechanism exists for Slim symbols is my big problem with that. I believe a suggestion to address this has been rejected in the past because it would affect backwards compatibility.

**Comment by six42** [comment]
I fully agree with you on the unification of the symbols this would be an important cleanup of the slim specification.
And I think with some more refactoring it can be achieved that a switch allows to select one or the other syntax to be used for a test run or a wiki page. The symbol part is extra tricky as it impacts the Wiki and the SlimService. The switch would be required in both components.

**Comment by amolenaar** [comment]
@six42: My first reaction to adding extra formatting options is that it makes things even more complicated. Having both variables with `${x}` and `$x` is already confusing and can lead to inconsistency as @fhoeben points out.

That said. Having the original and substituted values in separate tags seems like a nice addition, although using `detail` and `summary` does not have my preference. I would go for `<del>`/`<ins>`.

**Comment by fhoeben** [comment]
@amolenaar how would you use del/ins? As-if the variable name was deleted and actual value inserted? For me: having the variable seen as 'strike-through' does not appeal...

**Comment by six42** [comment]
Thinking again about my proposal I would correct it:
Much of what I proposed might be better defined on a cell level. Long output is not only caused by symbols but can happen in any cell which returns results.
